src/App.tsx

```diff
@@ -1,61 +1,139 @@
 import './App.scss';
+import React, { useState } from 'react';
+import usersFromServer from './api/users';
+import todosFromServer from './api/todos';
+import { User } from './Types/User';
+import { Todo } from './Types/Todo';
+import { TodoList } from './components/TodoList';
 
-// import usersFromServer from './api/users';
-// import todosFromServer from './api/todos';
+function getUser(userId: number): User | null {
+  const foundUser = usersFromServer.find(user => user.id === userId);
+
+  return foundUser || null;
+}
+
+export const initialTodos: Todo[] = todosFromServer.map(todo => ({
+  ...todo,
+  user: getUser(todo.userId),
+}));
+
+export const App: React.FC = () => {
+  const [title, setTitle] = useState('');
+  const [userId, setUserId] = useState(0);
+  const [isTitleError, setIsTitleError] = useState(false);
+  const [isUserError, setIsUserError] = useState(false);
+  const [newTodos, setNewTodos] = useState(initialTodos);
+
+  const reset = () => {
+    setUserId(0);
+    setTitle('');
+    setIsUserError(false);
+    setIsTitleError(false);
+  };
+
+  const addTodo = () => {
+    if (userId && title) {
+      const newId = Math.max(...newTodos.map(todo => todo.id)) + 1;
+
+      const newTodo = {
+        id: newId,
+        userId,
+        completed: false,
+        title,
+        user: getUser(userId),
+      };
+
+      setNewTodos((currentTodos) => ([...currentTodos, newTodo]));
+      reset();
+    }
+
+    if (!userId) {
+      setIsUserError(true);
+    }
+
+    if (!title) {
+      setIsTitleError(true);
+    }
+  };
+
+  const handleSubmit = (event: React.FormEvent<HTMLFormElement>) => {
+    event.preventDefault();
+    addTodo();
+  };
+
+  const handleTitle = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setTitle(event.target.value);
+    setIsTitleError(false);
+  };
+
+  const handleUserId = (event: React.ChangeEvent<HTMLSelectElement>) => {
+    setUserId(+event.target.value);
+    setIsUserError(false);
+  };
 
-export const App = () => {
   return (
     <div className="App">
       <h1>Add todo form</h1>
 
-      <form action="/api/users" method="POST">
+      <form
+        action="/api/users"
+        method="POST"
+        onSubmit={handleSubmit}
+      >
         <div className="field">
-          <input type="text" data-cy="titleInput" />
-          <span className="error">Please enter a title</span>
+          <label>
+            <input
+              type="text"
+              data-cy="titleInput"
+              placeholder="Enter a title"
+              value={title.replace(/[^a-z\d\s]+/gi, '').trim()}
```

Comment:
Avoid modifying the input value directly in the `value` prop. Instead, perform the sanitization inside the `handleTitle` function.